### PR TITLE
[#187082048] Add Django 4 compatibility

### PR DESCRIPTION
**Pull request by mrmikardo**
# Proposed changes

This PR adds Django 4 support to the Housekeep fork of [the original Zinnia project](https://github.com/Fantomas42/django-blog-zinnia), which has not been maintained for several years.

A summary of the changes made here is:

- Bump the Django version specified in the `setup.py` to be at least 4.0.0
- Cut the `django-tagging` dependency from `setup.py`: `django-tagging` is another unmaintained package which doesn't support Django 4. A [Housekeep-specific fork](https://github.com/HousekeepLtd/django-tagging) of this package has been created, which (i) supports Django 4 and (ii) fixes some other (maybe Django-4 related?) bugs.
- Fix some outdated imports in the demo project: this means it actually runs and can be used to test the project!
- Update some Python 2 specific code (object inheritance, `__future__` imports)

Note that the change to the `django-tagging` dependency means that from now on, we'll have to list that dependency in the main Housekeep repo's Pipfile.

### CODE REVIEW

File: demo/manage.py
Comment (mrmikardo) [inline demo/manage.py:1]: Added this file as using `django-admin` is annoying (you have to manually specify the settings module path).

File: demo/settings.py [hunk 1]
```diff
@@ -28,31 +28,6 @@
...
-    ('ca', gettext('Catalan')),
-    ('tr', gettext('Turkish')),
-    ('sv', gettext('Swedish')),
-    ('is', gettext('Icelandic')),
-    ('hr_HR', gettext('Croatian')),
-    ('pt_BR', gettext('Brazilian Portuguese')),
-    ('fa_IR', gettext('Persian')),
-    ('fi_FI', gettext('Finnish')),
-    ('uk_UA', gettext('Ukrainian')),
-    ('zh-hans', gettext('Simplified Chinese')),
```
Comment (mrmikardo) [inline demo/settings.py:55]: We don't support any of these languages, and they were causing issues when I bumped to Django 4: ✂️ .

File: zinnia/templates/zinnia/entry_list.html
Comment (mrmikardo) [inline zinnia/templates/zinnia/entry_list.html:1]: Changes in this file update a template tag which [has been deprecated](https://code.djangoproject.com/ticket/31532).

File: demo/settings.py (hunk 1 above)
Comment (horthbynorthwest) [reply demo/settings.py:55]: Makes sense, time to get them gone!

File: zinnia/tests/implementations/urls/custom_detail_views.py [hunk 2]
```diff
@@ -1,5 +1,4 @@
 """Test urls for the zinnia project"""
-from django.conf.urls import url
```
Comment (horthbynorthwest) [inline zinnia/tests/implementations/urls/custom_detail_views.py:2]: was this just a left over unused import, I can't see where this was used in this file?

File: zinnia/tests/implementations/urls/default.py [hunk 3]
```diff
@@ -1,6 +1,5 @@
 """Test urls for the zinnia project"""
 from django.conf.urls import include
-from django.conf.urls import url
```
Comment (horthbynorthwest) [inline zinnia/tests/implementations/urls/default.py:3]: same here

**Review by horthbynorthwest — CHANGES_REQUESTED** [review]
only additional thing we should do is bump the version like we fixed in #3 (and initially changed in #2)

**Review by horthbynorthwest — APPROVED** [review]
😎

**Pull request merged**